src/api/propensity-api.js

```diff
@@ -13,187 +13,7 @@
-   * - provide a way to start the payment flow for a specific offer.
-   *   (in this case, the next event will be ACTION_OFFER_SELECTED
-   *   or ACTION_PAYMENT_FLOW_STARTED depending on if that button
-   *   took the user to a checkout page on the publishers site or
-   *   directly started the payment flow).
-   * The field 'active' of PropensityEvent, which carries this
-   * event, must be set to true since this is a user action.
-   * The JSON block with this event can provide additional information
-   * such as the source, indicating what caused the user to navigate
-   * to this page.
-   * For example:
-   * const propensityEvent = {
-   *   name: 'subscriptions_landing_page',
-   *   active: true,
-   *   data: {'source': 'marketing_via_email'}
-   * }
-   */
-  ACTION_SUBSCRIPTIONS_LANDING_PAGE: 'subscriptions_landing_page',
-  /**
-   * ACTION_OFFER_SELECTED event.
-   * User has selected an offer.
-   * The field 'active' of PropensityEvent, which carries this
-   * event, must be set to true since this is a user action.
-   * The JSON block can provide the product selected.
-   * For example: {
-   *   name: 'offer_selected',
-   *   active: true,
-   *   data: {product': 'basic-monthly'}
-   * }
-   * When offer selection starts the payment flow directly,
-   * use the next event ACTION_PAYMENT_FLOW_STARTED instead.
-   */
-  ACTION_OFFER_SELECTED: 'offer_selected',
-  /**
-   * ACTION_PAYMENT_FLOW_STARTED event.
-   * User has started payment flow.
-   * The field 'active' of PropensityEvent, which carries this
-   * event, must be set to true since this is a user action.
-   * The JSON block can provide the product selected.
-   * For example:
-   * const propensityEvent = {
-   *   name: 'payment_flow_started',
-   *   active: true,
-   *   data: {product': 'basic-monthly'}
-   * }
-   */
-  ACTION_PAYMENT_FLOW_STARTED: 'payment_flow_start',
-  /**
-   * ACTION_PAYMENT_COMPLETED.
-   * User has made the payment for a subscription.
-   * The field 'active' of PropensityEvent, which carries this
-   * event, must be set to true since this is a user action.
-   * The JSON block can provide the product user paid for.
-   * For example:
-   * const propensityEvent = {
-   *   name: 'payment_complete',
-   *   active: true,
-   *   data: {product': 'basic-monthly'}
-   * }
-   */
-  ACTION_PAYMENT_COMPLETED: 'payment_complete',
-  /**
-   * EVENT_CUSTOM: custom publisher event.
-   * The field 'active' of PropensityEvent, which carries this
-   * event, must be set to true or false depending on if the event
-   * was generated as a result of a user action.
-   * The JSON block can provide the event name for the custom event.
-   * For example:
-   * const propensityEvent = {
-   *   name: 'custom',
-   *   active: true,
-   *   data: {
-   *     'event_name': 'social_share',
-   *     'platform_used': 'whatsapp'
-   *   }
-   *  }
-   */
-  EVENT_CUSTOM: 'custom',
-}
+import * as LoggerApi from './logger-api';
```

Comment:
Lets not delete this here, it serves as Propensity documentation

Reply:
Logger API must have event and SubscriptionState (currently defined in Propensity) in order to create functions that accept those enums.
Propensity must rely on Logger API so it can implement the interface declared in it
So if I do not move the primary definitions of Event and SubscriptionState to Logger API it creates a circular reference.

Redeclaring Event and SubscriptionState in Propensity is simply done so no one needs to change references to continue using SwG.

Reply:
I didn't mean create circular reference, I mean keep Propensity and Logger referes to Propensity API.

Reply:
Sorry, I left the word API out above.  Let me try to explain why you are asking for a circular reference again:  

1) PropensityAPI MUST reference LoggerAPI to implement its interface.  This cannot change.
2) LoggerAPI needs the enums currently defined in PropensityAPI because they are part of the interface it defines.  

If I do not move those enums out of Propensity API it is a circular reference.